tracer/src/Datadog.Trace/PlatformHelpers/ContainerMetadata.cs

```diff
@@ -66,6 +102,69 @@ public static string ParseCgroupLine(string line)
                        : null;
         }
 
+        /// <summary>
+        /// Uses regular expression to try to extract controller/cgroup-node-path pairs from the specified string
+        /// then, using these pairs, will return the first inode found from the concatenated path
+        /// <paramref name="controlGroupsMountPath"/>/controller/cgroup-node-path.
+        /// If no inode could be found, this will return <c>null</c>.
+        /// </summary>
+        /// <param name="controlGroupsMountPath">Path to the cgroup mount point.</param>
+        /// <param name="lines">Lines of text from a cgroup file.</param>
+        /// <returns>The cgroup node controller's inode if found; otherwise, <c>null</c>.</returns>
+        public static string ExtractInodeFromCgroupLines(string controlGroupsMountPath, IEnumerable<string> lines)
+        {
+            foreach (var line in lines)
+            {
+                var tuple = ParseControllerAndPathFromCgroupLine(line);
+                if (tuple is not null
+                    && !string.IsNullOrEmpty(tuple.Item2)
+                    && (tuple.Item1 == string.Empty || string.Equals(tuple.Item1, "memory", StringComparison.OrdinalIgnoreCase)))
+                {
+                    string controller = tuple.Item1;
+                    string cgroupNodePath = tuple.Item2;
+                    var path = Path.Combine(controlGroupsMountPath, controller, cgroupNodePath.TrimStart('/'));
+
+                    if (Directory.Exists(path) && TryGetInode(path, out long output))
+                    {
+                        return output.ToString();
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Uses regular expression to try to extract a controller/cgroup-node-path pair from the specified string.
+        /// </summary>
+        /// <param name="line">A single line from a cgroup file.</param>
+        /// <returns>The controller/cgroup-node-path pair if found; otherwise, <c>null</c>.</returns>
+        public static Tuple<string, string> ParseControllerAndPathFromCgroupLine(string line)
+        {
+            var lineMatch = Regex.Match(line, CgroupRegex);
+
+            return lineMatch.Success
+                       ? new(lineMatch.Groups[1].Value, lineMatch.Groups[2].Value)
+                       : null;
+        }
+
+        internal static bool TryGetInode(string path, out long result)
+        {
+            result = 0;
+            var context = SynchronizationContext.Current;
```

Comment:
you get the context but you don't do anything with it right? I guess the intention is to replace the context to avoid a deadlock or something?

Reply:
I started by copying and pasting the async code, but I don't need this any longer. I'll remove this

Reply:
Removed in 6e868d89d392f5eeb7e814f653f36b8cc043f945